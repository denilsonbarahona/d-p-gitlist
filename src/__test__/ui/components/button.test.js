import React from 'react'
import {mount} from "enzyme"
import Button from '../../../shared/components/button'
import {create} from 'react-test-renderer'


describe('button',()=>{
    const button = mount(<Button/>)

    test('render del button',()=>{
        expect(button.length).toEqual(1)
    })

    test('create snapshot',()=>{
        const buttonSnapshot = create(<Button />)
        expect(buttonSnapshot.toJSON()).toMatchSnapshot()
    })

    test('simulate event click',()=>{
        const onClickEvent = jest.fn();
        const buttonEvent = mount(<Button onClick={onClickEvent} />)
        buttonEvent.find('button').simulate('click');
        expect(onClickEvent).toHaveBeenCalledTimes(1);
    })
})