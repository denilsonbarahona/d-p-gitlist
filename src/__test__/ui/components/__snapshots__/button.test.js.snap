// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button create snapshot 1`] = `<button />`;
